File: TemplateTools/src/main/java/com/lh/component/common/PredictorResult.java
package com.lh.component.common;

import java.util.LinkedList;
import java.util.List;

public class PredictorResult {
    private final int MAX_STACK = 5;
    private final LinkedList<Pair<Double, String>> result = new LinkedList<>();

    public void addResult(String prediction, double avgDistance) {
        for (int i = 0; i < result.size(); i++) {
            if (avgDistance < result.get(i).first) {
                result.add(i, new Pair<>(avgDistance, prediction));
                break;
            }
        }
        if (result.size() < MAX_STACK)
            result.add(new Pair<>(avgDistance, prediction));
        verify();
    }

    public List<Pair<Double, String>> getResult() {
        return result;
    }

    public void remove(String s) {
        Pair<Double, String> x = null;
        for (int i = 0; i < result.size(); i++) {
            if (result.get(i).second.equals(s)) {
                x = result.get(i);
                break;
            }
        }
        if (x != null)
            result.remove(x);
    }

    private void verify() {
        while (result.size() > MAX_STACK)
            result.remove(result.size() - 1);
    }
}

File: proj.android/app/src/main/java/com/lh/kete/algorithm/predictor/PredictorResult.kt
package com.lh.kete.algorithm.predictor

import java.util.*

class PredictorResult {
    private val MAX_STACK = 5
    private val result = LinkedList<Pair<Float, String>>()
    fun addResult(predition: String, avgDistance: Float) {
        for (i in 0 until result.size) {
            if (avgDistance < result[i].first) {
                result.add(i, Pair(avgDistance, predition))
                break
            }
        }
        if (result.size < MAX_STACK)
            result.add(Pair(avgDistance, predition))
        verify()
    }

    fun getResult(): List<Pair<Float, String>> {
        return result
    }

    fun remove(item: String) {
        var needToRemove: Pair<Float, String>? = null
        result.forEach {
            if (it.second == item) {
                needToRemove = it
            }
        }
        needToRemove?.let {
            result.remove(it)
        }
    }

    private fun verify() {
        while (result.size > MAX_STACK)
            result.removeAt(result.size - 1)
    }
}